Sources/StreamVideo/Controllers/CallController.swift

```diff
@@ -601,33 +593,32 @@ class CallController: @unchecked Sendable {
     }
 
     private func subscribeToParticipantsCountUpdatesEvent(_ call: Call?) {
-        participantsCountUpdatesTask?.cancel()
-        participantsCountUpdatesTask = nil
+        disposableBag.remove("subscribeToParticipantsCountUpdatesEvent")
 
         guard let call else { return }
 
-        participantsCountUpdatesTask = Task {
-            let anonymousUserRoleKey = "anonymous"
-            for await event in call.subscribe(for: CallSessionParticipantCountsUpdatedEvent.self) {
-                Task { @MainActor in
-                    call.state.participantCount = event
-                        .participantsCountByRole
-                        .filter { $0.key != anonymousUserRoleKey } // TODO: Workaround. To be removed
-                        .values
-                        .map(UInt32.init)
-                        .reduce(0) { $0 + $1 }
-
-                    // TODO: Workaround. To be removed
-                    if event.anonymousParticipantCount > 0 {
-                        call.state.anonymousParticipantCount = UInt32(event.anonymousParticipantCount)
-                    } else if let anonymousCount = event.participantsCountByRole[anonymousUserRoleKey] {
-                        call.state.anonymousParticipantCount = UInt32(anonymousCount)
-                    } else {
-                        call.state.anonymousParticipantCount = 0
-                    }
+        let anonymousUserRoleKey = "anonymous"
+
+        call
+            .eventPublisher(for: CallSessionParticipantCountsUpdatedEvent.self)
+            .sinkTask { @MainActor [weak call] event in
+                call?.state.participantCount = event
+                    .participantsCountByRole
+                    .filter { $0.key != anonymousUserRoleKey } // TODO: Workaround. To be removed
+                    .values
+                    .map(UInt32.init)
+                    .reduce(0) { $0 + $1 }
+
+                // TODO: Workaround. To be removed
```

Comment:
this never got fixed on the backend, right?

Reply:
I haven't been informed about any changes here tbh. Worth checking after this PR